### PR TITLE
VideoPress: track upload error handler

### LINKED ISSUE #33117: Jetpack VideoPress: Text track uploader doesn't handle network errors
### Impacted plugin

VideoPress

### Quick summary

Older versions of the VideoPress block would show you an error if the .vtt file being updloaded to the tracks/captions settings was invalid:

Screenshot 2023-09-15 at 9 15 55 AM

This no longer occurs and the .vtt file is "added" but actually doesn't exists on the server. If you refresh the block the .vtt entry will disappear from the tracks settings.

It should be fairly simple to fix, this line here should not naively accept `src` as the text track but instead inspect it for a network error object:

https://github.com/Automattic/jetpack/blob/trunk/projects/packages/videopress/src/client/block-editor/blocks/video/components/tracks-control/index.tsx#L129 

### Steps to reproduce

* Get a .vtt file, but make it invalid by removing the `WEBVTT` header text.
* Attach the .vtt to a video.
* You will see the track get added, now refresh the page.
* The .vtt will disappear because it was rejected at the API endpoint.

### A clear and concise description of what you expected to happen.

_No response_

### What actually happened

_No response_

### Impact

All

### Available workarounds?

No but the platform is still usable

### Platform (Simple and/or Atomic)

Atomic, Self-hosted

### Logs or notes

_No response_

### PR DESCRIPTION
**Pull request by CGastrell**
Add error handling for .vtt track files

Fixes https://github.com/Automattic/jetpack/issues/33117

## Proposed changes:
This PR adds error handling to the .vtt track file upload process. Whenever an upload/validation error occurs, it will show an error sign (was already there, just not used for upload errors):

image

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Does this pull request change what data or activity we track or use?
No

## Testing instructions:
Insert a VideoPress block (upload or use an existing video). Upload a track file. See that the process works as usual.
Try uploading a malformed .vtt file, it should show an error. 
Canceling and opening the upload modal again should allow restarting the process.
Once an upload has failed, the "Save" button won't become available again unless you load a new track on the file input.

Valid .vtt:
```
WEBVTT

1
00:00:00.000 --> 00:00:03.667
(upbeat instrumental music)

2
00:00:05.020 --> 00:00:09.000
<v ->Hey there!</v>

3
00:00:09.000 --> 00:00:12.420
I'm joined here with Ron Kendig
```

Invalid .vtt
```
xxWEBVTT

1
00:00:00.000 --> 00:00:03.667
(upbeat instrumental music)

2
00:00:05.020 --> 00:00:09.000
<v ->Hey there!</v>

3
00:00:09.000 --> 00:00:12.420
I'm joined here with Ron Kendig
```

### CODE REVIEW

File: projects/packages/videopress/changelog/fix-videopress-tracks-error-handler [hunk 1]
```diff
@@ -0,0 +1,4 @@
+Significance: patch
+Type: added
+
+Add error handling for .vtt utrack files upload process
```
Comment (jgcaruso) [inline projects/packages/videopress/changelog/fix-videopress-tracks-error-handler]: small typo? or is `utrack` something?
```suggestion
Add error handling for .vtt track files upload process
```

**Review by roundhill — DISMISSED** [review]
Tested well for me, I have one UX suggestion:

If you upload a track that fails, I think the error should clear out if you select another track file again. You kind of get stuck in the error state unless you close the modal and open it again.

**Comment by CGastrell** [comment]
> Tested well for me, I have one UX suggestion:
> 
> If you upload a track that fails, I think the error should clear out if you select another track file again. You kind of get stuck in the error state unless you close the modal and open it again.

That should work, as long as you select a different file(name) 😬 The change event doesn't trigger if the file has the same name, sorry to say. Going beyond that would involve actually reading the file and check for differences with the previous one. It should be fine if it's not big, but I don't know what would happen with bigger files (file content should be set in the state).

**Comment by roundhill** [comment]
> That should work, as long as you select a different file(name) 😬 The change event doesn't trigger if the file has the same name, sorry to say. Going beyond that would involve actually reading the file and check for differences with the previous one. It should be fine if it's not big, but I don't know what would happen with bigger files (file content should be set in the state).

Ah, I see. I was using the same file name (Was just removing the `WEBVTT` header to make it invalid. Probably not a big deal!

**Review by roundhill — DISMISSED** [review]
Tested on wp.com and working better - but I'm getting an eternal `Save` state now after it successfully adds the track - do you see the same?

Screenshot 2023-09-22 at 2 26 46 PM

**Comment by jgcaruso** [comment]
With the latest code on wpcom, when uploading an "Invalid VTT" file, the uploader just disappears instead of showing the error message.

**Comment by CGastrell** [comment]
> With the latest code on wpcom, when uploading an "Invalid VTT" file, the uploader just disappears instead of showing the error message.

Seeing the same thing. I’ll let you know when the PR is ready for review, for now I’m figuring out the different error handling on the environments and pushing to create new builds. Thanks!

**Review by jgcaruso — APPROVED** [review]
Validated in wpcom and jetpack. The Track validation error now appears as expected when uploading an invalid vtt track.

**Pull request merged**